src/lib/bootloader/stage1_proposal.rb

```diff
@@ -0,0 +1,256 @@
+          log.info "/boot is on logical partition or separated and uses btrfs, mbr is preferred"
+          selected_location = :mbr
+        end
+
+        if !stage1.can_use_boot?
+          log.info "/boot cannot be used to install stage1"
+          selected_location = :mbr
+        end
+
+        assign_bootloader_device(selected_location)
+      end
+
+      def separated_boot?
+        Yast::BootStorage.BootPartitionDevice != Yast::BootStorage.RootPartitionDevice
+      end
+
+      def extended_partition
+        init_boot_info unless @boot_initialized
+
+        @extended
+      end
+
+      def logical_boot?
+        init_boot_info unless @boot_initialized
+
+        @logical_boot
+      end
+
+      def boot_with_btrfs?
+        init_boot_info unless @boot_initialized
+
+        @boot_with_btrfs
+      end
+
+      def init_boot_info
+        return if @boot_initialized
+
+        @boot_initialized = true
+        boot_disk_map = Yast::Storage.GetTargetMap[Yast::BootStorage.disk_with_boot_partition] || {}
+        boot_part = Yast::Storage.GetPartition(Yast::Storage.GetTargetMap,
+          Yast::BootStorage.BootPartitionDevice)
+        @logical_boot = boot_part["type"] == :logical
+        @boot_with_btrfs = boot_part["used_fs"] == :btrfs
+
+        # check for sure also underlaying partitions
+        (boot_disk_map["partitions"] || []).each do |p|
+          @extended = p["device"] if p["type"] == :extended
+          next unless underlaying_boot_partition_devices.include?(p["device"])
+
+          @boot_with_btrfs = true if p["used_fs"] == :btrfs
+          @logical_boot = true if p["type"] == :logical
+        end
+      end
+
+      # determine the underlaying devices for the "/boot" partition (either the
+      # BootPartitionDevice, or the devices from which the soft-RAID device for
+      # "/boot" is built)
+      def underlaying_boot_partition_devices
+        ::Bootloader::Stage1Device.new(Yast::BootStorage.BootPartitionDevice).real_devices
+      end
+
+      def boot_partition_on_mbr_disk?
+        underlaying_boot_partition_devices.any? do |dev|
+          pdp = Yast::Storage.GetDiskPartition(dev)
+          p_disk = pdp["disk"] || ""
+          p_disk == Yast::BootStorage.mbr_disk
+        end
+      end
+    end
+
+    # s390x specific stage1 proposal
+    class S390 < Stage1Proposal
+      def propose
+        # s390 do not need any partition as it is stored to predefined zipl location
+        assign_bootloader_device(:none)
+
+        stage1.activate = false
+        stage1.generic_mbr = false
+      end
+    end
+
+    # ppc64le specific stage1 proposal
+    class PPC < Stage1Proposal
+      def propose
+        partition = proposed_prep_partition
+        if partition
+          assign_bootloader_device([:custom, partition])
+
+          stage1.activate = true
+          stage1.generic_mbr = false
+        # handle diskless setup, in such case do not write boot code anywhere
```

Comment:
actually it is correct as it is comment for below else branch and not for above lines